tools/sigclient/pkg/query/validation.go

```diff
@@ -0,0 +1,664 @@
+		}
+		bucket.MeasureVal, isMap = measure["MeasureVal"].(map[string]interface{})
+		if !isMap {
+			return fmt.Errorf("CreateExpResultForStats: measureVal is not a map, received type: %T", measure["MeasureVal"])
+		}
+		res.MeasureResults = append(res.MeasureResults, bucket)
+	}
+
+	return nil
+}
+
+func CreateResultForGroupBy(res *Result, response map[string]interface{}) error {
+	var err error
+	_, exist := response["bucketCount"]
+	if !exist {
+		return fmt.Errorf("bucketCount not found in response")
+	}
+	bucketCount, isFloat := response["bucketCount"].(float64)
+	if !isFloat {
+		return fmt.Errorf("CreateExpResultForGroupBy: bucketCount is not numeric, received type: %T", response["bucketCount"])
+	}
+	res.BucketCount = int(bucketCount)
+	res.GroupByCols, err = CreateListOfString(response["groupByCols"])
+	if err != nil {
+		return fmt.Errorf("CreateExpResultForGroupBy: Error fetching groupByCols from response, err: %v", err)
+	}
+
+	return CreateResultForStats(res, response)
+}
+
+func GetStringValueFromResponse(resp map[string]interface{}, key string) (string, error) {
+	value, exist := resp[key]
+	if !exist {
+		return "", fmt.Errorf("key %v not found in response", key)
+	}
+
+	query, isString := value.(string)
+	if !isString {
+		return "", fmt.Errorf("value %v is not a string, is of type %T", value, value)
+	}
+
+	return query, nil
+}
+
+func ValidateUniqueKeyColsInResult(res *Result) error {
+	if len(res.UniqueKeyCols) == 0 {
+		return fmt.Errorf("ValidateUniqueKeyColsInResult: UniqueKeyCols not found in result")
+	}
+	for _, record := range res.Records {
+		for _, col := range res.UniqueKeyCols {
+			_, exist := record[col]
+			if !exist {
+				return fmt.Errorf("ValidateUniqueKeyColsInResult: col %v not found in record: %v, UniqueKeyCols: %v", col, record, res.UniqueKeyCols)
+			}
+		}
+	}
+	return nil
+}
+
+func ReadAndValidateQueryFile(filePath string) (string, *Result, error) {
+	file, err := os.Open(filePath)
+	if err != nil {
+		return "", nil, fmt.Errorf("ReadAndValidateQueryFile: Error opening file: %v, err: %v", filePath, err)
+	}
+	defer file.Close()
+
+	decoder := json.NewDecoder(file)
+	response := make(map[string]interface{})
+	err = decoder.Decode(&response)
+	if err != nil {
+		return "", nil, fmt.Errorf("ReadAndValidateQueryFile: Error decoding file: %v, err: %v", filePath, err)
+	}
+
+	query, err := GetStringValueFromResponse(response, "queryText")
+	if err != nil {
+		return "", nil, fmt.Errorf("ReadAndValidateQueryFile: Error getting queryText from response, file: %v, err: %v", filePath, err)
+	}
+
+	if _, exist := response["expectedResult"]; !exist {
+		return "", nil, fmt.Errorf("ReadAndValidateQueryFile: expectedResult not found in file: %v", filePath)
+	}
+
+	expectedResult, isMap := response["expectedResult"].(map[string]interface{})
+	if !isMap {
+		return "", nil, fmt.Errorf("ReadAndValidateQueryFile: expectedResult is not valid it's of type: %T, file: %v", response["expectedResult"], filePath)
```

Comment:
nit: add punctuation between "not valid" and "it's of type"

Reply:
addressed in the latest commit.